corehq/messaging/fcm_util.py

```diff
@@ -0,0 +1,74 @@
+import logging
+import os
+
+from django.conf import settings
+from firebase_admin import messaging, credentials, initialize_app
+
+logger = logging.getLogger(__name__)
+
+MAX_DEVICES_ALLOWED_MULTICAST = 500
+HQ_FCM_UTIL = None
+
+
+def is_fcm_available():
+    if not settings.FCM_CREDS_PATH:
+        return False
+    return os.path.isfile(settings.FCM_CREDS_PATH)
+
+
+class FCMUtilException(Exception):
+    pass
+
+
+class DevicesLimitExceeded(FCMUtilException):
+    def __init__(self):
+        super().__init__(f"Max devices allowed is {MAX_DEVICES_ALLOWED_MULTICAST}! Please execute in batches.")
+
+
+class FCMUtil:
+    def __init__(self, app_name, creds_path=settings.FCM_CREDS_PATH):
+        creds = credentials.Certificate(creds_path)
```

Comment:
do we need to check for `is_fcm_available` here before this is run?

Reply:
or raise an exception if creds missing.

Reply:
@mkangia
Resolved https://github.com/dimagi/commcare-hq/commit/40fc5b619e4ac6202030448ae3661c7a5e52b327